packages/template-retail-react-app/app/routes.jsx

```diff
@@ -118,9 +129,88 @@ export const routes = [
     }
 ]
 
-export default () => {
+const componentNameMap = {
+    'Home': Home,
+    'Cart': Cart,
+    'ProductDetail': ProductDetail,
+    'ProductList': ProductList,
+    'PageNotFound': PageNotFound
+}
```

Comment:
I believe we are using this map to deserialize the serialized route configuration, there is probably a better way to do this without having to manually maintain this mapping object if you want to support other page types. 

If we don't plan on supporting other page types and this is a finite list, then it would be nice to hide this object from plan view, maybe it even becomes configurable if possible.